File: tests/sdk/go/secrets_kms_test.go
package sdk_test

import (
	"encoding/base64"
	"encoding/json"
	"net/http"
	"net/url"
	"testing"
)

func TestSecretManagerCreateAccessSmoke(t *testing.T) {
	ep := requireReady(t)
	token := requireToken(t)
	project := projectID()
	secretID := uniqueID("sdk-secret")
	base := ep + "/v1/projects/" + project + "/secrets/" + secretID

	status, body := doJSON(t, http.MethodPost, ep+"/v1/projects/"+project+"/secrets?secretId="+url.QueryEscape(secretID), token, map[string]any{
		"replication": map[string]any{"automatic": map[string]any{}},
	})
	if status != http.StatusOK {
		t.Fatalf("create secret status=%d body=%s", status, body)
	}
	t.Cleanup(func() {
		_, _, _ = doJSONErr(http.MethodDelete, base, token, nil)
	})

	payload := base64.StdEncoding.EncodeToString([]byte("sdk-smoke"))
	status, body = doJSON(t, http.MethodPost, base+":addVersion", token, map[string]any{
		"payload": map[string]any{"data": payload},
	})
	if status != http.StatusOK {
		t.Fatalf("addVersion status=%d body=%s", status, body)
	}

	status, body = doJSON(t, http.MethodGet, base+"/versions/latest:access", token, nil)
	if status != http.StatusOK {
		t.Fatalf("access secret status=%d body=%s", status, body)
	}
	var accessed map[string]any
	if err := json.Unmarshal(body, &accessed); err != nil {
		t.Fatalf("decode access: %v body=%s", err, body)
	}
	payloadObj, _ := accessed["payload"].(map[string]any)
	gotB64, _ := payloadObj["data"].(string)
	got, err := base64.StdEncoding.DecodeString(gotB64)
	if err != nil {
		t.Fatalf("decode payload data: %v body=%s", err, body)
	}
	if string(got) != "sdk-smoke" {
		t.Fatalf("payload=%q want sdk-smoke body=%s", got, body)
	}
}

func TestListKMSKeyRingsSmoke(t *testing.T) {
	ep := requireReady(t)
	token := requireToken(t)
	project := projectID()

	path := ep + "/v1/projects/" + project + "/locations/global/keyRings"
	status, body := doJSON(t, http.MethodGet, path, token, nil)
	if status != http.StatusOK {
		t.Fatalf("list KMS keyRings status=%d body=%s", status, body)
	}
	var parsed map[string]any
	if err := json.Unmarshal(body, &parsed); err != nil {
		t.Fatalf("decode KMS keyRings: %v body=%s", err, body)
	}
	if _, ok := parsed["keyRings"]; !ok {
		t.Fatalf("missing keyRings field body=%s", body)
	}
}

File: tests/sdk/nodejs/secrets_kms.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import { doJSON, hasOwn, projectID, requireReady, requireToken, uniqueID } from "./helpers.mjs";

test("secret manager create access smoke", async (t) => {
  const ep = await requireReady(t);
  if (!ep) return;
  const token = requireToken(t);
  if (!token) return;
  const project = projectID();
  const secretID = uniqueID("sdk-secret");
  const base = `${ep}/v1/projects/${project}/secrets/${secretID}`;

  const created = await doJSON(
    "POST",
    `${ep}/v1/projects/${project}/secrets?secretId=${encodeURIComponent(secretID)}`,
    token,
    { replication: { automatic: {} } },
  );
  assert.equal(created.status, 200, `create secret status=${created.status} body=${created.body}`);
  t.after(async () => {
    try {
      await doJSON("DELETE", base, token);
    } catch {
      /* best-effort cleanup */
    }
  });

  const payload = Buffer.from("sdk-smoke", "utf8").toString("base64");
  const added = await doJSON("POST", `${base}:addVersion`, token, {
    payload: { data: payload },
  });
  assert.equal(added.status, 200, `addVersion status=${added.status} body=${added.body}`);

  const accessed = await doJSON("GET", `${base}/versions/latest:access`, token);
  assert.equal(accessed.status, 200, `access secret status=${accessed.status} body=${accessed.body}`);
  const parsed = JSON.parse(accessed.body);
  const got = Buffer.from(parsed.payload?.data || "", "base64").toString("utf8");
  assert.equal(got, "sdk-smoke", accessed.body);
});

test("list kms key rings smoke", async (t) => {
  const ep = await requireReady(t);
  if (!ep) return;
  const token = requireToken(t);
  if (!token) return;
  const project = projectID();

  const { status, body } = await doJSON(
    "GET",
    `${ep}/v1/projects/${project}/locations/global/keyRings`,
    token,
  );
  assert.equal(status, 200, `list KMS keyRings status=${status} body=${body}`);
  const parsed = JSON.parse(body);
  assert.ok(hasOwn(parsed, "keyRings"), body);
});
